### PR TITLE
Video listing & single video page

### PR DESCRIPTION
**Pull request by Varun-28**
###Done with Video Listing and Single Video Page

*Files to check:*

- video-context.js
- SingleVideo.js
- VideoListing.js

*Features Added:*

- Single Video page with the current video 
- Single Video page with related videos
- Video Listing page
- Category options in the video listing page
- Everything is responsive
- Everything follow light and dark mode theme

Video Listing Page Desktop View:
Screenshot 2022-05-19 at 15-39-18 CraftTube

Video Listing Page Mobile View:
Screenshot 2022-05-19 at 15-40-08 CraftTube

Single Video Page Desktop View:
Screenshot 2022-05-19 at 15-40-28 CraftTube

Single Video Page Mobile View:
Screenshot 2022-05-19 at 15-40-45 CraftTube

Light mode:
Screenshot 2022-05-19 at 15-41-12 CraftTube

### CODE REVIEW

**Review by shilpe26 — APPROVED** [review]
🔹UI looks awesome
🔹Code is properly intended
🔹Good to merge

File: src/App.js [hunk 1]
```diff
@@ -47,47 +50,55 @@ function App() {
...
-          />
+          <Route path="/video" element={<Video />}>
+            <Route index path="videolisting" element={<VideoListing />} />
+            <Route path="videolisting/:videoId" element={<SingleVideo />} />
+            <Route
+              path="watchlater"
+              element={
+                <RequiresAuth>
+                  <WatchLater />
+                </RequiresAuth>
```
Comment (Hamdrive) [inline src/App.js:61]: You can explore having nested conditional routes, thus not having to do repetition

**Review by Hamdrive — COMMENTED** [review]
- Really like the UI of the video library, also the features given to users.
- For suggestion in Drawer component, you can explore dynamically rendering the nav components using `map()` ensuring that you are reducing repetition of components

File: src/App.js (hunk 1 above)
Comment (Varun-28) [reply src/App.js:61]: OK, sure I will do the suggested changes

**Pull request merged**